weed/s3api/s3api_object_copy_handlers.go

```diff
@@ -309,3 +310,7 @@ func processMetadataBytes(reqHeader http.Header, existing map[string][]byte, rep
 
 	return
 }
+
+func IsS3FileEntry(entry *filer_pb.Entry) bool {
+	return !entry.IsDirectory || (entry.Attributes != nil && entry.Attributes.Mime != "")
```

Comment:
s3 directories has mime as "httpd/unix-directory"

Reply:
Why is mime of s3 directory set to "httpd/unix-directory"? directory is only a logical concept in s3 api design
Here is to consider how to correctly handle s3 files when entry is both s3 files and directories, because directories do not have mime, and s3 files are generally mime (if not, they need to be set to the default application/octorstream). In this way, when an entry is a directory, we can know whether the entry is a file at the same time through mime.

Reply:
@kmlebedev any comments about setting the mime of s3 directory to "httpd/unix-directory"?